core/src/avm2/globals/flash/display/graphics.rs

```diff
@@ -103,14 +103,18 @@ pub fn begin_gradient_fill<'gc>(
         let gradient_type = args.get_string(activation, 0)?;
         let gradient_type = parse_gradient_type(activation, gradient_type)?;
         let colors = args.get_object(activation, 1, "colors")?;
-        let alphas = args.get_object(activation, 2, "alphas")?;
-        let ratios = args.get_object(activation, 3, "ratios")?;
+        let alphas = args.try_get_object(activation, 2);
+        let ratios = args
+            .try_get_object(activation, 3)
+            .expect("Ratios should never be null");
```

Comment:
FP also permits null here (I think it just calculates the ratios to be equally spaced then).

Reply:
I tested and the behavior is pretty strange; it seems like the start of the first gradient is further back than the start of the fill/line. I left it like this so it can be reported it if it ever happens. I'll add a comment for clarity.